### PR TITLE
Do not set feature test macro in public header

### PR DESCRIPTION
**Pull request by horgh**

### CODE REVIEW

File: t/bad_databases_t.c [hunk 1]
```diff
@@ -1,5 +1,8 @@
 // This test currently does not work on Windows as nftw is
 // not available.
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#endif
```
Comment (oschwald) [inline t/bad_databases_t.c:5]: It doesn't matter too much, but could we have just set these in `maxminddb_test_helper.h`?

**Review by oschwald — CHANGES_REQUESTED** [review]
One optional improvement. This or even an internal header throughout might be a bit cleaner, but I don't have strong opinion.

Do you think we will need to update any of our extensions for this change? If so, we should be sure to release those before we release this.

**Comment by horgh** [comment]
I tested the Python, PHP, and Perl extensions with this version of libmaxminddb installed. I ran the tests as shown in the GitHub test action for each. All ran without issue, so I think we're okay with the extensions as they are.

**Comment by horgh** [comment]
Console output of the tests - https://gist.github.com/horgh/f6c8b7a5ba3b042fcccb4111481bf214

**Review by oschwald — APPROVED** [review]

**Pull request merged**